pkg/ebpf/c/common/buffer.h

```diff
@@ -351,6 +351,12 @@ statfunc int events_perf_submit(program_data_t *p, u32 id, long ret)
     u32 size = sizeof(event_context_t) + sizeof(u8) +
                p->event->args_buf.offset; // context + argnum + arg buffer size
 
+    // Check, at the last moment, if the policies version changed during the event processing.
+    // If so, skip the event submission as it is considered stale.
```

Comment:
Previous snapshots would only work for buffered events already sent in an end-to-end policy version window. It turns out that as the event is composed of different steps/programs, if the version is different, the rules that correspond to that event may be wrong in sequential steps/programs, so I think that in the final phase if the version is different, it is not safe to use this event. In other words: it was matched/composed in more than one rule windows.

Reply:
Rethinking it, if we keep it hot in a go map, the uit64 field with the right policies version address would be coerced during the whole pipeline instead of re-retrieved.

It would work unlike the event pointer pool which stores unused pointers, this map would store used policy pointers to prevent them from being freed. It might work. LMKWYT.

Reply:
As we discuss and decide offline, we will always submit the event and leave it to the user to fetch the corresponding policies version object.